Client/UserResearch/OnboardingUserResearch.swift

```diff
@@ -4,25 +4,80 @@
 
 import Foundation
 import Leanplum
+import Shared
 
 struct LPVariables {
     static var showOnboardingScreen = LPVar.define("showOnboardingScreen", with: true)
 }
 
+enum OnboardingScreenType: String {
+    case versionV1 // Default
+    case versionV2 // New version 2
+}
+
 class OnboardingUserResearch {
-    // Delegate closure
+    // Closure delegate
     var updatedLPVariables: ((LPVar?) -> Void)?
     // variable
     var lpVariable: LPVar?
+    // Constants
+    private let onboardingScreenTypeKey = "onboardingScreenTypeKey"
+    // Saving user defaults
+    private let defaults = UserDefaults.standard
+    // Publicly accessible onboarding screen type
+    var onboardingScreenType:OnboardingScreenType? {
+        set(value) {
+            if value == nil {
+                defaults.removeObject(forKey: onboardingScreenTypeKey)
+            } else {
+                defaults.set(value?.rawValue, forKey: onboardingScreenTypeKey)
+            }
+        }
+        get {
+            guard let value = defaults.value(forKey: onboardingScreenTypeKey) else {
+                return nil
+            }
+            return OnboardingScreenType(rawValue: value as! String) ?? nil
+        }
+    }
     
-    // Initializer
+    // MARK: Initializer
     init(lpVariable: LPVar? = LPVariables.showOnboardingScreen) {
         self.lpVariable = lpVariable
     }
     
+    // MARK: public
     func lpVariableObserver() {
         Leanplum.onVariablesChanged {
             self.updatedLPVariables?(self.lpVariable)
         }
     }
+    
+    func updateValue(value: Bool) {
+        // For LP variable below is the convention
+        // we are going to follow
+        // True = Current Onboarding Screen
+        // False = New Onboarding Screen
+        onboardingScreenType = value ? .versionV1 : .versionV2
+    }
+    
+    func updateTelemetry() {
+        print("lp variant \(String(describing: Leanplum.variants()))")
+        if Leanplum.variants() != nil {
+            let lpVariants = Leanplum.variants()?.first
+            if let lpData = lpVariants as? Dictionary<String, AnyObject> {
+                var abTestId = ""
```

Comment:
`let  abTestId = lpData["abTestId"] as? Int64 ?? ""` ?

Reply:
I tried this but since **lpData["abTestId"] as? Int64** gives us an int and we are saying if its nil then make it a String the LHS gets confused and we get an error like below.  

<img width="709" alt="image" src="https://user-images.githubusercontent.com/8919439/81967978-c26c0880-95e9-11ea-9a6c-e6145d2b9445.png">

I can do something like below but again what advantage would it give us over what we have. If the idea is to get rid of **if let** then yeah we can go with b) 

What do you think? 🤔 

a) What we currently have:

```
var abTestId = ""
if let value = lpData["abTestId"] as? Int64 {
      abTestId = "\(value)"
}
```

b) New version: 
```
let abTestIdValue = lpData["abTestId"] as? Int64
let abTestId: String = abTestIdValue == nil ? "" : "\(abTestIdValue)"
```

Reply:
Also version b) makes produces optional string so we either to force unwrap or provide default value

Reply:
I just see multi-line init of a var and assume it could be one-line let syntax without thinking it through. Sorry to send you down the garden path there! The original was fine.

Reply:
Np at all, happy to address and improve code where possible 😊